e2e/test/scripts/create_cluster.sh

```diff
@@ -3,39 +3,63 @@
 set -o errexit
 set -o pipefail
 set -o nounset
+set -x
 
-export LINODE_API_TOKEN="$1"
+export LINODE_TOKEN="$1"
 export CLUSTER_NAME="$2"
-export IMAGE="$3"
-export K8S_VERSION="$4"
+export IMG="$3"
+export KUBERNETES_VERSION="$4"
+export CAPL_VERSION="0.1.0"
+export WORKER_MACHINE_COUNT=2
+export LINODE_CONTROL_PLANE_MACHINE_TYPE=g6-standard-2
+export LINODE_MACHINE_TYPE=g6-standard-2
+export KUBECONFIG="$(realpath "$(dirname "$0")/../kind-management.conf")"
+export ROOT_DIR="$(git rev-parse --show-toplevel)"
 
 if [[ -z "$5" ]]
 then
-  export REGION="eu-west"
+  export LINODE_REGION="eu-west"
 else
-  export REGION="$5"
+  export LINODE_REGION="$5"
 fi
 
-cat > cluster.tf <<EOF
-variable "nodes" {
-  default = 2
-}
+(cd ${ROOT_DIR}/deploy ; set +x ; ./generate-manifest.sh ${LINODE_TOKEN} ${LINODE_REGION})
 
-module "k8s" {
-  source       = "git::https://github.com/linode/terraform-linode-k8s.git"
-  k8s_version  = "${K8S_VERSION}"
-  linode_token = "${LINODE_API_TOKEN}"
-  ccm_image    = "${IMAGE}"
-  region       = "${REGION}"
-  cluster_name = "${CLUSTER_NAME}"
-  nodes        = var.nodes
-}
-EOF
+kubectl create ns ${CLUSTER_NAME}
+(cd $(realpath "$(dirname "$0")"); clusterctl generate cluster ${CLUSTER_NAME} \
+  --target-namespace ${CLUSTER_NAME} \
+  --flavor clusterclass-kubeadm \
+  --config clusterctl.yaml \
+  | kubectl apply --wait -f -)
 
-terraform workspace new ${CLUSTER_NAME}
+c=8
+until kubectl get secret -n ${CLUSTER_NAME} ${CLUSTER_NAME}-kubeconfig; do
```

Comment:
more of a nit, but rather than looping through here to wait for the kubeconfig to be available, you can use `kubectl wait` instead to wait for the cluster resource to be `ready` and the kubeconfig to exist like
```
kubectl wait cluster --timeout 10m --for condition=ready ${CLUSTER_NAME}
```

Reply:
Doesn't work without `sleep`